k2/csrc/ragged_ops.cu

```diff
@@ -1143,7 +1144,32 @@ Array1<int32_t> GetTransposeReordering(Ragged<int32_t> &src, int32_t num_cols) {
   if (device_type == kCpu) return GetTransposeReorderingCpu(src, num_cols);
 
   K2_CHECK_EQ(device_type, kCuda);
+  (void)GetTransposeReorderingThreeAxesCuda;  // remove compiler warnings
 
+#if 1
+  int32_t num_buckets = num_cols;
+  int32_t num_elements = src.values.Dim();
+  int32_t log_buckets = static_cast<int32_t>(ceilf(log2f(num_buckets)));
+
+  Array1<int32_t> tmp_values(context, num_elements);
+  Array1<int32_t> ans = Range(context, num_elements, 0);
+
+  cudaStream_t stream = context->GetCudaStream();
+
+  size_t temp_storage_bytes = 0;
+  K2_CUDA_SAFE_CALL(cub::DeviceRadixSort::SortPairs(
+      nullptr, temp_storage_bytes, src.values.Data(), tmp_values.Data(),
+      ans.Data(), ans.Data(), num_elements, 0, log_buckets, stream));
+
+  Array1<int8_t> d_temp_storage(context, temp_storage_bytes);
```

Comment:
Not sure if it works to pass `ans.Data()` as `tmp_values` above and allocate d_temp_storage and tmp_values together within in an Array1...though it may looks strange but save a memory allocation.

Reply:
Forget it, current code is much cleaner.

Reply:
sounds reasonable, fixed, thanks

Reply:
Ah, thanks, maybe add documentation to say why we do this, i.e. save one memory allocation.